### PR TITLE
Enable config for release artifacts

### PR DESCRIPTION
**Pull request by sebastienvas**

### CODE REVIEW

File: bin/init.sh [hunk 1]
```diff
@@ -9,7 +19,7 @@ if [ $PDIR != "$GOPATH/src/istio.io/pilot" ]; then
 fi
 
 # Building and testing with Bazel
-bazel build //...
+bazel build ${bazel_build_args[@]} //...
```
Comment (kyessenov) [inline bin/init.sh]: If you really need to customize bazel build, then I suggest remove this line from this script.

File: bin/push-docker [hunk 2]
```diff
@@ -37,19 +37,21 @@ function docker_push() {
 hubs="gcr.io/istio-testing"
 local_tag=$(whoami)_$(date +%y%m%d_%h%m%s)
 tags="${local_tag}"
+bazel_build_args=(--output_groups=static)
 
 while [[ $# -gt 0 ]]; do
     case "$1" in
         -tag) tags="$2"; shift ;;
         -hub) hubs="$2"; shift ;;
+        -config) bazel_build_args+=(--config="$2"); shift ;;
```
Comment (kyessenov) [inline bin/push-docker]: can we call it bazelconfig or something more specific than "config"?

File: bin/upload-istioctl [hunk 3]
```diff
@@ -7,19 +7,22 @@
 set -o errexit
 set -o nounset
 set -o pipefail
+set -ex
```
Comment (kyessenov) [inline bin/upload-istioctl]: why? that's same as options above

File: tools/bazel.rc [hunk 4]
```diff
@@ -2,5 +2,8 @@
 build --verbose_failures
 test --test_output=errors
 
+# Release Build
+build:release -c opt
```
Comment (kyessenov) [inline tools/bazel.rc]: does -c opt even have any effect on golang?

File: bin/upload-istioctl [hunk 5]
```diff
@@ -7,19 +7,22 @@
...
+INIT_ARGS=()
 
 function cleanup {
   rm -rf "${BIN_DIR}"
 }
 
-while getopts :p:r arg; do
+while getopts :c:p:r arg; do
   case ${arg} in
+    c) INIT_ARGS+=(-config ${OPTARG});;
```
Comment (kyessenov) [inline bin/upload-istioctl]: can we avoid mysterious short flags and use same flag as above?

File: tools/bazel.rc (hunk 4 above)
Comment (sebastienvas) [inline tools/bazel.rc]: From @ldemailly it does !

File: bin/init.sh (hunk 1 above)
Comment (sebastienvas) [inline bin/init.sh]: Removing and building where ?

File: bin/upload-istioctl (hunk 3 above)
Comment (sebastienvas) [inline bin/upload-istioctl]: I ll remove it, this was just for debugging to print the commands.

File: tools/bazel.rc (hunk 4 above)
Comment (kyessenov) [inline tools/bazel.rc]: what changes in the go binary?

Comment (ldemailly) [inline tools/bazel.rc]: the 2x benefits I measured is with envoy (c++) but I expect debug to be -gcflags="-N -l" and opt to be without those and thus make a difference, not sure the bazel go rules do do that (we should check, but even if they don't today, having 2 or more config will be useful in the future)

Comment (kyessenov) [inline tools/bazel.rc]: yes, envoy is special. let's first check if there's any difference with rules_go. if not, then we can't even use bazel to accomplish optimized build.

Comment (ldemailly) [inline tools/bazel.rc]: I asked in https://github.com/bazelbuild/rules_go/issues/741 but we should use this config seb is added anyway, to change the behavior of kube-inject / to select the optimized sidecar images for instance

Comment (sebastienvas) [inline tools/bazel.rc]: I wish I had known that before starting.

File: bin/push-docker (hunk 2 above)
Comment (sebastienvas) [inline bin/push-docker]: Done

File: bin/upload-istioctl (hunk 3 above)
Comment (sebastienvas) [inline bin/upload-istioctl]: Done

File: bin/upload-istioctl (hunk 5 above)
Comment (sebastienvas) [inline bin/upload-istioctl]: Done

File: bin/init.sh (hunk 1 above)
Comment (sebastienvas) [inline bin/init.sh]: I am not sure what you meant here.

**Comment by sebastienvas** [comment]
Closing this as this is does not have any impact on the artifacts.

**Comment by ldemailly** [comment]
it will in next bazel: see https://github.com/bazelbuild/rules_go/pull/743

**Comment by kyessenov** [comment]
keep the PR open until rules_go releases the support.

**Comment by ldemailly** [comment]
or we could just get it in and it'll get benefits automatically once bazel side changes - whichever works for you

**Comment by sebastienvas** [comment]
closing this as I just force pushed to this branch and lost the code for this :(

**Comment by ldemailly** [comment]
the bazel change is in, can we at least keep this for early part of 0.3

**Comment by sebastienvas** [comment]
we can but I lost the code, so I have to it all over again.

**Comment by ldemailly** [comment]
code is here no ? https://patch-diff.githubusercontent.com/raw/istio/pilot/pull/1063.patch

**Comment by ldemailly** [comment]
or use the reflog to recover the code ?
(but we can close this if you prefer - and just leave the issue)

**Comment by sebastienvas** [comment]
The code that you pointed out refer to another PR. Tell me more about
reflog.

**Comment by ldemailly** [comment]
you can do "git reflog" and see your local history and go back to any point in that history
using git reset using the id on the left

**Comment by sebastienvas** [comment]
it's gone!

**Pull request closed without merging**